script/tool/lib/src/build_examples_command.dart

```diff
@@ -99,38 +99,60 @@ class BuildExamplesCommand extends PackageLoopingCommand {
   Future<PackageResult> runForPackage(Directory package) async {
     final List<String> errors = <String>[];
 
+    final bool isPlugin = isFlutterPlugin(package);
     final Iterable<_PlatformDetails> requestedPlatforms = _platforms.entries
         .where(
             (MapEntry<String, _PlatformDetails> entry) => getBoolArg(entry.key))
         .map((MapEntry<String, _PlatformDetails> entry) => entry.value);
-    final Set<_PlatformDetails> buildPlatforms = <_PlatformDetails>{};
-    final Set<_PlatformDetails> unsupportedPlatforms = <_PlatformDetails>{};
-    for (final _PlatformDetails platform in requestedPlatforms) {
-      if (pluginSupportsPlatform(platform.pluginPlatform, package)) {
-        buildPlatforms.add(platform);
-      } else {
-        unsupportedPlatforms.add(platform);
-      }
-    }
+
+    // Platform support is checked at the package level for plugins; there is
+    // no package-level platform information for non-plugin packages.
+    final Set<_PlatformDetails> buildPlatforms = isPlugin
+        ? requestedPlatforms
+            .where((_PlatformDetails platform) =>
+                pluginSupportsPlatform(platform.pluginPlatform, package))
+            .toSet()
+        : requestedPlatforms.toSet();
     if (buildPlatforms.isEmpty) {
       final String unsupported = requestedPlatforms.length == 1
           ? '${requestedPlatforms.first.label} is not supported'
-          : 'None of [${requestedPlatforms.map((_PlatformDetails p) => p.label).join(',')}] are supported';
+          : 'None of [${requestedPlatforms.map((_PlatformDetails p) => p.label).join(', ')}] are supported';
```

Comment:
nit: I ran into this too, it probably would be better to pull the join argument to a const or wrap that behavior in a function so it could be updated in one place.

Reply:
I pulled out a local function for the two lines that were doing the same display list transform.